pkg/controllers/machine/liveness.go

```diff
@@ -16,30 +16,73 @@ package machine
 
 import (
 	"context"
+	"time"
 
+	"github.com/prometheus/client_golang/prometheus"
+	"go.uber.org/multierr"
 	"k8s.io/utils/clock"
+	"knative.dev/pkg/logging"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 
 	"github.com/aws/karpenter-core/pkg/apis/settings"
 	"github.com/aws/karpenter-core/pkg/apis/v1alpha5"
+	"github.com/aws/karpenter-core/pkg/metrics"
+	"github.com/aws/karpenter-core/pkg/utils/result"
 )
 
 type Liveness struct {
 	clock      clock.Clock
 	kubeClient client.Client
 }
 
+// creationTTL is a heuristic time that we expect to succeed with our cloudprovider.Create() call
+// If we don't succeed within this time, then we should delete and try again through some other mechanism
+const creationTTL = time.Minute * 2
```

Comment:
Can we make this something longer like 5-10 minutes? How often are we retrying? Is it possible that in normal success situations that this would take longer than 2 minutes?

Reply:
No, in normal success situations, this should take like 10s. It really depends for how long we think we should hang provisioning/deprovisioning while we are failing to launch. To me 2m seems like enough time. What makes you want to extend it?